include/oneapi/dpl/pstl/glue_algorithm_ranges_impl.h

```diff
@@ -602,6 +752,246 @@ struct __merge_fn
 
 inline constexpr __internal::__merge_fn merge;
 
+namespace __internal
+{
+
+struct __fill_fn
+{
+    template<typename _ExecutionPolicy, std::ranges::random_access_range _R, typename _T = std::ranges::range_value_t<_R>>
+    requires oneapi::dpl::is_execution_policy_v<std::remove_cvref_t<_ExecutionPolicy>> &&
+             std::ranges::output_range<_R, const _T&> && std::ranges::sized_range<_R>
+
+    std::ranges::borrowed_iterator_t<_R>
+    operator()(_ExecutionPolicy&& __exec, _R&& __r, const _T& __value) const
+    {
+        const auto __dispatch_tag = oneapi::dpl::__ranges::__select_backend(__exec);
+        return oneapi::dpl::__internal::__ranges::__pattern_fill(__dispatch_tag, std::forward<_ExecutionPolicy>(__exec),
+            std::forward<_R>(__r), __value);
+    }
+}; //__fill_fn
+
+} //__internal
+
+inline constexpr __internal::__fill_fn fill;
+
+namespace __internal
+{
+
+struct __generate_fn
+{
+    template<typename _ExecutionPolicy, std::ranges::random_access_range _R, std::copy_constructible _F>
+    requires oneapi::dpl::is_execution_policy_v<std::remove_cvref_t<_ExecutionPolicy>> && std::invocable<_F&>
+        && std::ranges::output_range<_R, std::invoke_result_t<_F&>> && std::ranges::sized_range<_R>
```

Comment:
same comment for `std::ranges::output_range`

Reply:
Update: see an answer in the thread above.